tardis/opacities/opacity_solver.py

```diff
@@ -59,13 +61,54 @@ def solve(self, legacy_plasma) -> OpacityState:
         else:
             tau_sobolev = calculate_sobolev_line_opacity(
                 atomic_data.lines,
-                legacy_plasma.level_number_density,
-                legacy_plasma.time_explosion,
-                legacy_plasma.stimulated_emission_factor,
+                plasma.level_number_density,
+                plasma.time_explosion,
+                plasma.stimulated_emission_factor,
+            )
+
+        opacity_state = OpacityState.from_legacy_plasma(plasma, tau_sobolev)
+
+        return opacity_state
+
+    def solve(self, plasma) -> OpacityState:
+        """
+        Solves the opacity state
+
+        Parameters
+        ----------
+        plasma : tarids.plasma.BasePlasma
+            legacy base plasma
+
+        Returns
+        -------
+        OpacityState
+        """
+        atomic_data = plasma.atomic_data
```

Comment:
Seems a bit odd to define this atomic_data variable considering all the other properties that are just used from the plasma in lines 92 and 102 onward. 

I also don't know if doing this for atom data would make a pointer or a copy, but if it's a copy it could be a substantial memory bump if you have large atomic data.